lib/src/postgrest.dart

```diff
@@ -57,16 +61,19 @@ class PostgrestClient {
     FetchOptions options = const FetchOptions(),
   }) {
     final url = '${this.url}/rpc/$fn';
-    return PostgrestRpcBuilder(url,
-            headers: headers,
-            schema: schema,
-            httpClient: httpClient,
-            options: options,
-            isolate: _isolate)
-        .rpc(params, options);
+    return PostgrestRpcBuilder(
+      url,
+      headers: headers,
+      schema: schema,
+      httpClient: httpClient,
+      options: options,
+      isolate: _isolate,
+    ).rpc(params, options);
   }
 
-  Future<void> dispose() {
-    return _isolate.dispose();
+  Future<void> dispose() async {
+    if (!_hasCustomIsolate) {
+      return _isolate.dispose();
+    }
```

Comment:
Love how you have made it so that custom isolates can be reused even after Supabase instance has been disposed!